docs/traffic-control.md

```diff
@@ -0,0 +1,239 @@
+can be `Mirror` or `Redirect`. For `Mirror` action, `targetPort` must be
+specified to the port to which the traffic will be mirrored to. For `Redirect`
+action, both `targetPort` and `returnPort` need to be specified, the latter of
+which represents a port from which the traffic will be sent back to OVS and be
+forwarded to its original destination.
+
+### TargetPort
+
+The `targetPort` field specifies the port to which the traffic should be
+redirected or mirrored. There are five kinds of ports that can be used to receive
+mirrored traffic:
+
+**ovsInternal**: This specifies an OVS internal port on all Nodes. A Pod's
+traffic will be redirected or mirrored to the OVS internal port on the same Node
+that hosts the Pod. The port doesn't need to exist in advance, Antrea will
+create the port if it doesn't exist. To use an OVS internal port, `name` of the 
+port must be provided:
+
+```yaml
+ovsInternal:
+  name: tap0
+```
+
+**device**: This specifies a network device on all Nodes. A Pod's traffic will
+be redirected or mirrored to the network device on the same Node that hosts the
+Pod. The network device must exist on all Nodes and Antrea will attach it to the
+OVS bridge if not already attached. To use a network device, `name` of the
+device must be provided:
+
+```yaml
+device:
+  name: eno2
+```
+
+**geneve**: This specifies a remote destination based on GENEVE tunnel. All
+selected Pods' traffic will be redirected or mirrored to the destination via
+a GENEVE tunnel. The `remoteIP` field must be provided to specify the IP address
+of the destination. Optionally, the `destinationPort` field could be used to
+specify the UDP destination port of the tunnel, or 6081 will be used by default.
+If Virtual Network Identifier (VNI) is desired, the `vni` field can be specified
+to an integer in the range 0-16,777,215:
+
+```yaml
+geneve:
+  remoteIP: 10.0.10.2
+  destinationPort: 6081
+  vni: 1
+```
+
+**vxlan**: This specifies a remote destination based on VXLAN tunnel. All
+selected Pods' traffic will be redirected or mirrored to the destination via
+a VXLAN tunnel. The `remoteIP` field must be provided to specify the IP address
+of the destination. Optionally, the `destinationPort` field could be used to
+specify the UDP destination port of the tunnel, or 4789 will be used by default.
+If Virtual Network Identifier (VNI) is desired, the `vni` field can be specified
+to an integer in the range 0-16,777,215:
+
+```yaml
+vxlan:
+  remoteIP: 10.0.10.2
+  destinationPort: 4789
+  vni: 1
+```
+
+**gre**: This specifies a remote destination based on GRE tunnel. All
+selected Pods' traffic will be redirected or mirrored to the destination via
+a GRE tunnel. The `remoteIP` field must be provided to specify the IP address
+of the destination. If GRE key is desired, the `key` field can be specified to
+an integer in the range 0-4,294,967,295:
+
+```yaml
+gre:
+  remoteIP: 10.0.10.2
+  key: 1
+```
+
```

Comment:
I haven't added ERSPAN yet, will test that configuration and get more understanding about its usage first.

Reply:
I wonder if we will need to support IPsec as well for this. Sorry if it has been discussed already.

Reply:
It hasn't been discussed. Thanks for the suggestion. I think it's a good idea to support IPsec, which is suitable for the case in which traffic is mirrored to a destination via unsecured network.

Reply:
Created #3834 to track the proposal

Reply:
I feel we can see if any feedback before adding IPsec support. Not saying it wont be useful, but I want to control the fast growing feature set and code base.